File: 038_count_and_say/Solution.java
class Solution {
    public String countAndSay(int n) {
        StringBuilder tmp1 = new StringBuilder("1");
        StringBuilder tmp2;
        for (int i = 2; i < n + 1; i++) {
            tmp2 = new StringBuilder();
            int count = 1;
            for (int j = 1; j < tmp1.length(); j++) {
                if (tmp1.charAt(j) == tmp1.charAt(j - 1)) count++;
                else tmp2.append(count).append(tmp1.charAt(j - 1));
            }
            tmp2.append(count).append(tmp1.charAt(tmp1.length() - 1));
            tmp1 = tmp2;
        }

        return tmp1.toString();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.countAndSay(4));
    }
}

File: 038_count_and_say/Solution.py
class Solution:
    def countAndSay(self, n: int) -> str:
        tmp1 = '1'
        for i in range(2, n + 1):
            tmp2 = ''
            count = 1
            for j in range(1, len(tmp1)):
                if tmp1[j] == tmp1[j - 1]: count += 1
                else:
                    tmp2 += str(count) + tmp1[j - 1]
                    count = 1
            tmp2 += str(count) + tmp1[-1]
            tmp1 = tmp2

        return tmp1


if __name__ == '__main__':
    solution = Solution()
    print(solution.countAndSay(4))

